berserk/clients/bulk_pairings.py

```diff
@@ -0,0 +1,91 @@
+"""Bulk pairings client"""
+from __future__ import annotations
+
+from typing import Any, cast, Dict, List, Optional
+
+from .base import BaseClient
+from ..enums import Variant
+from ..formats import JSON_LIST, JSON
+from ..types.bulk_pairings import BulkPairingGame
+from ..types.core.aliases import ID
+
+
+class BulkPairings(BaseClient):
+    """Client for bluk pairing related endpoints."""
+
+    def view_upcoming(self) -> List[BulkPairingGame]:
+        """View upcoming bulk pairings.
+
+        :return: List of upcoming bulk pairings.
+        :rtype: List[BulkPairingGame]
+        """
+        path: str = "/api/bulk-pairing"
+        return cast(List[BulkPairingGame], self._r.get(path, fmt=JSON_LIST))
+
+    def create(
+        self,
+        players_tokens: List[str],
+        clock_limit: int,
+        clock_increment: int,
+        days: Optional[int] = None,
+        pair_at: Optional[int] = None,
+        start_clocks_at: Optional[int] = None,
+        rated: Optional[bool] = None,
+        variant: Optional[Variant] = None,
+        fen: Optional[str] = None,
+        message: Optional[str] = None,
+        rules: Optional[list[str]] = None,
```

Comment:
We use `x | None`, instead of `Optional[x]` elsewhere in the code

Reply:
`rules` should not by `str` but instead a list of literal union of all possible values, `"noAbort" "noRematch" "noGiveTime" "noClaimWin" "noEarlyDraw"`: https://lichess.org/api#tag/Bulk-pairings/operation/bulkPairingCreate

Reply:
Yeah, I'm reworking the enumerations. Will fix the Optional.